<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Flexible and Customized Logo Plots using symbols, alphabets, numbers and alphanumeric strings • Logolas</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Logolas</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="articles/index.html">Articles</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kkdey/Logolas">
    <span class="fa fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->


      </header><div class="row">
  <div class="contents col-md-9">
    <div id="logolas" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#logolas" class="anchor"> </a>Logolas</h1></div>

<p>A R package for Flexible logo plotting device for alphanumeric English symbols.</p>
<p><a href="http://kkdey.github.io/">Kushal K Dey</a>, <a href="http://stephenslab.uchicago.edu/">Matthew Stephens</a></p>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"> </a>Introduction</h2>
<p>Logolas requires the CRAN-R packages <code>grid</code>, <code>gridExtra</code> and <code>ggplot2</code>. Also the package builds on the template set by the Bioconductor package <code>seqLogo</code> which was aimed at displaying A, C, G, T sequence logo motifs. In some sense, this package may be considered a more flexible extension of the ‘seqLogo’ package.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"> </a>Installation</h2>
<p>The package will be soon put on CRAN. As of now, you can install the developmental version from Github</p>
<pre><code>library(devtools)
install_github('kkdey/Logolas')</code></pre>
<p>Then load the package with:</p>
<pre><code>library(Logolas)</code></pre>
</div>
<div id="application-of-logolas---sequence-motif-example" class="section level2">
<h2 class="hasAnchor">
<a href="#application-of-logolas---sequence-motif-example" class="anchor"> </a>Application of Logolas - sequence motif example</h2>
<p>For sequence motif design problems, <code>seqLogo</code> is a standard package that is used and it is very easy to use. Here is an example from the vignette of the <code>seqLogo</code> package.</p>
<pre><code>library(seqLogo)
mFile &lt;- system.file("Exfiles/pwm1", package="seqLogo")
m &lt;- read.table(mFile)
p &lt;- makePWM(m)
seqLogo(p)</code></pre>
<p><img src="articles/figures/seqLogo_sequence.png" alt="Logo Plot" height="300" width="700"></p>
<p><code>seqLogo</code> uses the <code>makePWM</code> function to convert the matrix into a class “pwm” object by assigning the base information to the rows. To use our package, the user on the other hand would just need to set the row names to the base letters and the column names to the position index as follows. This would be more flexible when the user wants to play around with the order of the row names and change the column names.</p>
<pre><code>mat &lt;- m
rownames(mat) &lt;- c("A", "C", "G", "T")
colnames(mat) &lt;- 1:8

logomaker(mat,
          cols= RColorBrewer::brewer.pal(dim(mat)[1],
          name ="Spectral"),
          frame_width = 1,
          ic.scale = TRUE,
          yscale_change=FALSE,
          xlab="position")</code></pre>
<p><img src="articles/figures/logomaker_sequence_1.png" alt="Logo Plot" height="300" width="700"></p>
<p>Just as in <code>seqLogo</code> , <code>Logolas</code> too has an <code>ic.scale</code> option to determine if the heights of the bars should be same or should be determined by the information criterion.</p>
<pre><code><a href="reference/logomaker.html">logomaker(mat,
          cols= RColorBrewer::brewer.pal(dim(mat)[1],
          name ="Spectral"),
          frame_width = 1,
          ic.scale = FALSE,
          yscale_change=FALSE,
          xlab="position")</a></code></pre>
<p><img src="articles/figures/logomaker_sequence_2.png" alt="Logo Plot" height="300" width="700"></p>
<p>In addition, the user can use the <code>yscale_change</code> option which adjusts for the scale of Y axis as per data and blows up the logo plot, making it easier to visualize.</p>
<pre><code><a href="reference/logomaker.html">logomaker(mat,
          cols= RColorBrewer::brewer.pal(dim(mat)[1],
          name ="Spectral"),
          frame_width = 1,
          ic.scale = TRUE,
          yscale_change=TRUE,
          xlab="position")</a></code></pre>
<p><img src="articles/figures/logomaker_sequence_3.png" alt="Logo Plot" height="300" width="700"></p>
<p>Besides, Logolas also lets you play with the colors of the symbols, choose your own <code>ic</code> function that can be used to determine the heights of the stacks, set titles, x-labels, y-labels and axis names as per the user choice and also choose the relative width of each column in the logo stack, the kind of flexibility one does not get with <code>seqLogo</code> package.</p>
<p><code>seqLogo</code> is constrained to only logos for $A$, $C$, $G$ and $T$, but <code>Logolas</code> has a much wider mix of symbols or logos, encompassing all English uppercase alphabets, numbers, punctuation marks and all combinations numbers, alphabets and punctuations. Additionally, it lets the user to set new symbols and add them to the mix. All this makes <code>Logolas</code> a pretty generic logo building tool that can be applied in many applications. A few of those applications are provided below.</p>
</div>
<div id="application-of-logolas---protein-motif-example" class="section level2">
<h2 class="hasAnchor">
<a href="#application-of-logolas---protein-motif-example" class="anchor"> </a>Application of Logolas - protein motif example</h2>
<p>Our package is flexible to use on protein motif data as well. For protein motifs, insead of base symbols, we have symbols for $20$ amino acids, each of which is assigned one English alphabet. A demo example of that is provided below.</p>
<pre><code>counts_mat &lt;- rbind(c(0, 0, 100, 1, 2),
                    c(4, 3, 30, 35, 2),
                    c(100, 0, 10, 2, 7),
                    rep(0,5),
                    c(4, 2, 3, 7, 70),
                    c(1, 8, 0, 60, 3),
                    rep(0, 5),
                    c(4, 2, 100, 1, 1),
                    c(12, 8, 16, 7, 20),
                    c(55, 0, 1, 0, 12),
                    rep(0,5),
                    c(rep(0,3), 20, 0),
                    rep(0,5),
                    c(0, 0, 30, 0, 22),
                    c(1, 0, 12, 3, 10),
                    rep(0,5),
                    c(0, 1, 0, 34, 1),
                    c(0, 1, 12, 35, 1),
                    c(0, 30, 1, 10, 2),
                    c(0, 1, 4, 100, 2))



rownames(counts_mat) &lt;- c("A", "R", "N", "D",
                          "C", "E", "Q", "G",
                          "H", "I", "L", "K",
                          "M", "F", "P", "S",
                          "T", "W", "Y", "V")

colnames(counts_mat) &lt;- c("Pos 1", "Pos 2", "Pos 3", "Pos 4", "Pos 5")

cols1 &lt;- c(rev(RColorBrewer::brewer.pal(12, "Paired"))[c(3,4,7,8,11,12,5,6,9,10)],
           RColorBrewer::brewer.pal(12, "Set3")[c(1,2,5,8,9)],
           RColorBrewer::brewer.pal(9, "Set1")[c(9,7)],
           RColorBrewer::brewer.pal(8, "Dark2")[c(3,4,8)])
</code></pre>
<p>We then build the logo plots on this counts data.</p>
<pre><code><a href="reference/logomaker.html">logomaker(counts_mat,
          cols= cols1,
          frame_width = 1,
          ic.scale  = FALSE,
          yscale_change = FALSE)</a></code></pre>
<p><img src="articles/figures/logolas_protein_1.png" alt="Logo Plot" height="300" width="700"></p>
</div>
<div id="application-of-logolas---arxiv-category-logo" class="section level2">
<h2 class="hasAnchor">
<a href="#application-of-logolas---arxiv-category-logo" class="anchor"> </a>Application of Logolas - arXiv category logo</h2>
<p>Suppose we want to build a logo plot of the field categories of manuscipts submitted by authors on aRxiv. Here is a demo example on 4 Professors from Statistics department, University of Chicago where I come from. Note here the field categories are a combination of numbers and alphabets and not just a single alphabet as in the examples above. <code>Logolas</code> can be easily used even in such cases, with the same syntax structure as before.</p>
<p>We generate the counts data of number of publications for each category for each of the 4 Professors (representing column blocks here).</p>
<pre><code>library(aRxiv)
rec1 &lt;- arxiv_search('au:"Matthew Stephens"', limit=50)
rec2 &lt;- arxiv_search('au:"John Lafferty"', limit=50)
rec3 &lt;- arxiv_search('au:"Wei Biao Wu"', limit=50)
rec4 &lt;- arxiv_search('au:"Peter Mccullagh"', limit=50)

primary_categories_1 &lt;- toupper(rec1$primary_category)
primary_categories_2 &lt;- toupper(rec2$primary_category)
primary_categories_3 &lt;- toupper(rec3$primary_category)
primary_categories_4 &lt;- toupper(rec4$primary_category)

factor_levels &lt;- unique(c(unique(primary_categories_1),
                   unique(primary_categories_2),
                   unique(primary_categories_3),
                   unique(primary_categories_4)))

primary_categories_1 &lt;- factor(primary_categories_1, levels=factor_levels)
primary_categories_2 &lt;- factor(primary_categories_2, levels=factor_levels)
primary_categories_3 &lt;- factor(primary_categories_3, levels=factor_levels)
primary_categories_4 &lt;- factor(primary_categories_4, levels=factor_levels)


tab_data &lt;- cbind(table(primary_categories_1),
                  table(primary_categories_2),
                  table(primary_categories_3),
                  table(primary_categories_4))

colnames(tab_data) &lt;- c("Matthew Stephens",
                        "John Lafferty",
                        "Wei Biao Wu",
                        "Peter McCullagh")

tab_data &lt;- as.matrix(tab_data)</code></pre>
<p>We apply the <code>logomaker</code> function to plot the logo plots.</p>
<pre><code><a href="reference/logomaker.html">logomaker(tab_data,
          cols= RColorBrewer::brewer.pal(dim(tab_data)[1],
          name = "Spectral"),
          frame_width = 1,
          ic.scale = TRUE,
          pop_name = "arXiv field categories of UChicago STAT professors",
          xlab = "Professors",
          ylab = "Information content")
</a></code></pre>
<p><img src="articles/figures/logolas_arxiv_1.png" alt="Logo Plot" height="300" width="700"></p>
<pre><code><a href="reference/logomaker.html">logomaker(tab_data,
          cols= RColorBrewer::brewer.pal(dim(tab_data)[1],
          name = "Spectral"),
          frame_width = 1,
          ic.scale = FALSE,
          pop_name = "arXiv field categories of UChicago STAT professors",
          xlab = "Professors",
          ylab = "Information content")
</a></code></pre>
<p><img src="articles/figures/logolas_arxiv_2.png" alt="Logo Plot" height="300" width="700"></p>
</div>
<div id="application-of-logolas---mutational-profile" class="section level2">
<h2 class="hasAnchor">
<a href="#application-of-logolas---mutational-profile" class="anchor"> </a>Application of Logolas - mutational profile</h2>
<p>Sometimes a symbol or logo might be even more complex. For example, one may want to represent mutational profile as a logo plot, with the base substitution at the center and the flanking base information to the left and right of the substitution. For the substitution, one may want to use symbols like $ C -&gt; T$ etc. <code>Logolas</code> is flexible enough to mix symbols with arrow signs (among other signs) and it is very easy to define the logo as $C&gt;T$, $A&gt;G$ etc and $&gt;$ is interpreted as a right arrow (Similarly $&lt;$ is interpreted as left arrow). Check the example below</p>
<pre><code>library(seqLogo)
mFile &lt;- system.file("Exfiles/pwm1", package="seqLogo")
m &lt;- read.table(mFile)
p &lt;- makePWM(m)
mat1 &lt;- cbind(p@pwm[,c(3,4)], rep(0,4), p@pwm[,c(5,6)]);
colnames(mat1) &lt;- c("-2", "-1", "0", "1", "2")
mat2 &lt;- cbind(rep(0,6), rep(0,6),
              c(0.5, 0.2, 0.2, 0.05, 0.05, 0),
              rep(0,6), rep(0,6))
rownames(mat2) &lt;- c("C&gt;T", "C&gt;A", "C&gt;G",
                    "T&gt;A", "T&gt;C", "T&gt;G")

table &lt;- rbind(mat1, mat2)</code></pre>
<p>Now we apply Logolas to build the mutational logo plot.</p>
<pre><code><a href="reference/logomaker.html">logomaker(table,
          cols= rev(RColorBrewer::brewer.pal(dim(table)[1],
          name = "Spectral")),
          frame_width = 1,
          ic.scale = TRUE,
          yscale_change=TRUE,
          xlab = "Position",
          ylab = "Information content")
</a></code></pre>
<p><img src="articles/figures/mutational_profile_logolas.png" alt="Logo Plot" height="300" width="700"></p>
</div>
<div id="application-of-logolas---histone-marks" class="section level2">
<h2 class="hasAnchor">
<a href="#application-of-logolas---histone-marks" class="anchor"> </a>Application of Logolas - histone marks</h2>
<p>In studies related to histone marks, one might be interested to see if certain histone marks are prominent than others in some cell lines or tissues or in some genomic regions. In this case, we apply Logolas on an example data from Koch et al (2007) [Supp Table 2 of that paper]. The authors recorded number of histone modification sites identified by the HMM which overlap with an intergenic sequence, intron, exon, gene start and gene end for the lymphoblastoid cell line, GM06990, in the ChIP-CHIP data. Logolas provides a handy visualization to see how the patterns of histone modification sites changes across genomic region types for that cell line.</p>
<p>First we input the data from Supp Table 2 due to Koch et al (2007).</p>
<pre><code>mat &lt;- rbind(c(326, 296, 81, 245, 71),
             c(258, 228, 55, 273, 90),
             c(145, 121, 29, 253, 85),
             c(60, 52, 23, 180, 53),
             c(150, 191, 63, 178, 63))

rownames(mat) &lt;- c("H3K4ME1", "H3K4ME2", "H3K4ME3", "H3AC", "H4AC")
colnames(mat) &lt;- c("Intergenic",
                   "Intron",
                   "Exon \n 1000 KB window",
                   "Gene start \n 1000 KB window",
                   "Gene end \n 1000 KB window")
</code></pre>
<p>Now we apply Logolas on this matrix.</p>
<pre><code><a href="reference/logomaker.html">logomaker(mat,
          cols= sample(RColorBrewer::brewer.pal(10,name = "Spectral"), dim(mat)[1]),
          frame_width = 1,
          ic.scale = TRUE,
          pop_name = "Histone marks prop. abundance in various genomic regions",
          xlab = "",
          ylab = "Information content",
          yscale_change = TRUE,
          col_line_split = "black")</a></code></pre>
<p><img src="articles/figures/histone_marks.png" alt="Logo Plot" height="300" width="700"></p>
</div>
<div id="make-your-own-logo" class="section level2">
<h2 class="hasAnchor">
<a href="#make-your-own-logo" class="anchor"> </a>Make your own logo</h2>
<p>The biggest advantage of using this package is that an user can build his own atom logos (symbols, letters, punctuation, number) and add it to the set of logos already available in the default version of the package.</p>
<p>For example, if one wants to have the symbol Lambda as part of his logo, he can create it as follows</p>
<pre><code>LAMBDAletter &lt;- function(colfill="green"){

  x &lt;- c(0.15, 0.5, 0.85, 0.75, 0.5, 0.25)
  y &lt;- c(0, 1, 0, 0, 0.8, 0)

  fill &lt;- colfill
  id &lt;- rep(1, length(x))

  ll &lt;- list("x"= x,
             "y"= y,
             "id" = id,
             "fill" = fill)
  return(ll)
}</code></pre>
<p>Do make sure, your function name is of the form <code>*letter"</code> where you can be creative with the <code>"*"</code> part. Also make sure the name you put must be in uppercase letters. You can check if the symbol plot looks like a lambda or not.</p>
<pre><code>lambda &lt;- LAMBDAletter()
grid::grid.newpage()
grid::pushViewport(grid::viewport(x=0.5,y=0.5,width=1, height=1,
                                  clip=TRUE))
grid::grid.polygon(lambda$x, lambda$y,
                     default.unit="native",
                     id=lambda$id,
                     gp=grid::gpar(fill=lambda$fill,
                                   lwd=10))
  </code></pre>
<p><img src="articles/figures/lambda_plot.png" alt="Logo Plot" height="200" width="200"></p>
<p>Okay, once you are happy with the shape of the symbol, you can add it to the mix of all other symbols using the following command.</p>
<pre><code><a href="reference/makemylogo.html">makemylogo("AC&gt;EF/LAMBDA/W35",
            addlogos="LAMBDA",
            addlogos_text="LAMBDA",
            plot=TRUE)</a></code></pre>
<p><img src="articles/figures/makemylogo_with_lambda.png" alt="Logo Plot" height="300" width="700"></p>
<p>Note that we put <code>lambda</code> inside “/…/” to make sure that the function reads it as a new symbol and not general English alphabets or numbers. Once you run this, you should see the symbol as follows</p>
<p>This confirms that the symbol has been read into the mix, and now it can be used for stacking logo symbols under the <code><a href="reference/logomaker.html">logomaker()</a></code> functionality.</p>
<pre><code>counts_mat &lt;- rbind(c(0, 10, 100, 60, 20),
                    c(40, 30, 30, 35, 20),
                    c(100, 0, 15, 25, 75),
                    c(10, 30, 20, 50, 70)
)

colnames(counts_mat) &lt;- c("Pos 1", "Pos 2", "Pos 3", "Pos 4", "Pos 5")
rownames(counts_mat) &lt;- c("R/LAMBDA/Q", "A", "X", "Y")
counts_mat</code></pre>
<p>LAMBDA symbol is added under <code>addlogos</code> and <code>addlogos_text</code> options for the <code>logomaker</code> mix of symbols.</p>
<pre><code><a href="reference/logomaker.html">logomaker(counts_mat,
          cols= RColorBrewer::brewer.pal(dim(counts_mat)[1],
          name = "Spectral"),
          frame_width = 1,
          addlogos="LAMBDA",
          addlogos_text="LAMBDA")
          </a></code></pre>
<p><img src="articles/figures/lambda_logomaker.png" alt="Logo Plot" height="300" width="700"></p>
</div>
<div id="licenses" class="section level2">
<h2 class="hasAnchor">
<a href="#licenses" class="anchor"> </a>Licenses</h2>
<p>The Logolas package is distributed under [GPL - General Public License (&gt;= 2)]</p>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"> </a>Contact</h2>
<p>For any questions or comments, please contact <a href="kkdey@uchicago.edu" class="uri">kkdey@uchicago.edu</a></p>
</div>
<div id="acknowledgements" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgements" class="anchor"> </a>Acknowledgements</h2>
<p>We would like to acknowledge Oliver Bembom, the author of <code>seqLogo</code> for acting as an inspiration and giving us the base through his awesome package, on which we developed this software.</p>
<p>We would also like to thank Kevin Luo, Hussein -al Asadi, John Blischak and Alex White for helpful discussions.</p>
</div>
</div>
  </div>

  <div class="col-md-3" id="sidebar">
    <h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/kkdey/Logolas">https://​github.com/​kkdey/​Logolas</a>
</li>
</ul>
<h2>License</h2>
<p>GPL (&gt;= 2)</p>
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Kushal Dey <br><small class="roles"> Author, maintainer </small> </li>
</ul>
<html><body>
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/kkdey/Logolas"><img src="https://travis-ci.org/kkdey/Logolas.svg?branch=master" alt="Build Status"></a></li>
</ul>
</body></html>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Kushal Dey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
